public/app/features/dashboard/utils/getPanelMenu.ts

```diff
@@ -122,6 +122,20 @@ export function getPanelMenu(
 
   const menu: PanelMenuItem[] = [];
 
+  // Embedded dashboard's panels can only be edited
+  if (dashboard.meta.isEmbedded) {
+    if (dashboard.canEditPanel(panel) && !panel.isEditing) {
+      menu.push({
+        text: t('panel.header-menu.edit', `Edit`),
+        iconClassName: 'edit',
+        onClick: onEditPanel,
+        shortcut: 'e',
+      });
+    }
+
+    return menu;
+  }
```

Comment:
I have removed the kiosk mode references as I see it can be confusing. What we want here is not really a kiosk mode anyways but to use Grafana capabilities to render a dashboard from JSON, and this Grafana instance should be embeddable via an iframe.